src/interval_sets.jl

```diff
@@ -521,3 +521,55 @@ function find_intersections_helper!(result, x, y, lt)
 
     return unique!.(result)
 end
+
+function find_intersections(
+    x::AbstractVector{<:AbstractInterval{T1,Closed,Closed}},
+    y::AbstractVector{<:AbstractInterval{T2,Closed,Closed}},
+) where {T1,T2}
+    # Strategy:
+    # two binary searches per interval `I` in `x`
+    # * identify the set of intervals in `y` that start during-or-after `I`
+    # * identify the set of intervals in `y` that stop before-or-during `I`
+    # * intersect them
+    starts = first.(y)
+    starts_perm = sortperm(starts)
+    starts_sorted = starts[starts_perm]
+
+    # Sneaky performance optimization (makes a huge difference!)
+    # Rather than sorting `stops` relative to `y`, we sort it relative to `starts`.
+    # This allows us to work in the "starts" frame of reference until the very end.
+    # In particular, when we intersect the sets of intervals obtained from starts and from stops,
+    # the "starts" set can be kept as a `UnitRange`, making the intersection *much* faster.
```

Comment:
Any reason for using double-quote for `starts`?
```suggestion
    # Sneaky performance optimization (makes a huge difference!)
    # Rather than sorting `stops` relative to `y`, we sort it relative to `starts`.
    # This allows us to work in the `starts` frame of reference until the very end.
    # In particular, when we intersect the sets of intervals obtained from starts and from stops,
    # the `starts` set can be kept as a `UnitRange`, making the intersection *much* faster.
```